app/src/main/java/chat/rocket/android/server/infraestructure/SharedPreferencesSettingsRepository.kt

```diff
@@ -0,0 +1,26 @@
+package chat.rocket.android.server.infraestructure
+
+import chat.rocket.android.infrastructure.LocalRepository
+import chat.rocket.android.server.domain.SettingsRepository
+import chat.rocket.core.internal.SettingsAdapter
+import chat.rocket.core.model.Value
+
+class SharedPreferencesSettingsRepository(private val localRespository: LocalRepository) : SettingsRepository {
+
+    private val adapter = SettingsAdapter()
+
+    override fun save(url: String, settings: Map<String, Value<Any>>) {
+        localRespository.save("$SETTINGS_KEY$url", adapter.toJson(settings))
+    }
+
+    override fun get(url: String): Map<String, Value<Any>>? {
+        val settings = localRespository.get("$SETTINGS_KEY$url")
+        settings?.let {
+            return adapter.fromJson(it)
+        }
+
+        return null
+    }
+}
+
+const val SETTINGS_KEY = "settings_"
```

Comment:
You could put  the `SETTINGS_KEY` at the top of the `SharedPreferencesSettingsRepository` class.

Reply:
Actually I think this belongs to a companion object... It is intended to be used just by this class...

Either way, I think it belongs to the bottom of the class.

Quoting the Kotlin Coding Convetions doc

>Class layout
Generally, the contents of a class is sorted in the following order:
-Property declarations and initializer blocks
-Secondary constructors
-Method declarations
-Companion object

Reply:
Nice! And we never put our companion objects bellow the method declarations :grin: